macleann/enchiridion-client
# Major changes to styling and code readability

# Changes across the board to make styling more consistent and to make more readable code

I made a ton of styling changes and cleaned up my code to make it more readable. To make the code more readable, I generally used Prettier on most components, which make the changes seem more drastic than they actually are.

### New Components:

within the new `utils` directory:
- `makePlaylistImage.js`: Created this component as the logic to make little images for playlists based on the episodes that they contain was getting quite lengthy. I also figured I'd maybe want to use it in other places on the site so why not break it into its own component?
- `useScreenSize.js`: I had this logic in almost every single component honestly, which is just bad practice, so I learned how to make it a custom hook instead! This handles observing the viewport's size and setting a boolean value for isMobile at 768 pixels in width that can be used to for ternary operators across the site to handle whether someone is viewing something in mobile or not.

within `episodes` directory:
- `Episodes.js`: Made this to make it more consistent with `seasons` directory and to reduce the size of the return statement in both `PlaylistDetail.js` and `SeasonDetail.js`

### Minor Changes:
- Removed a leftover autofocus property from the `Register` component that was causing it to default to last name before first name
- Added content to the `Home` component as a placeholder
- Removed logic that got moved to `useScreenSize` component and imported the new custom hook to `PlaylistForm.js`
- Added a bunch of custom Tailwind class to `index.css`

### Major Changes:

`episodes/EpisodeDetail.js`:
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Defined `playlistId` via `useParams` hook
- Defined a new function that displays a link that will return the user to the previous page they were on depending on the previous page
- Defined a new function that displays the episode's air date as a more readable string
- Updated the return statement classNames and called new functions

`nav/NavBar.js`:
- Removed the unused local user variables
- Imported and added a `useEffect` hook to watch for changes in `isOpen` state and updates the a height property of that element
- Added an id to the topmost `<div>` so the above `useEffect` can update the height property accordingly

`playlists/PlaylistDetail.js`:
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Imported new `Episodes` component
- Added a link that takes the user back to 'Playlists' component
- Changed the return statement's classNames and called new `Episodes` component

`playlists/Playlists.js`:
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Imported new `makePlaylistImage` component
- Removed logic that is now handled by `isMobile`
- Removed logic that is now handled by `makePlaylistImage`
- Changed return statement's classNames to make it more consistent and mobile friendly

`search/SearchBar.js`:
- Removed unused `useNavigate` hook
- Added className logic that sticks the search bar just below the `NavBar` component based on the height property set by that component's new `useEffect`
- Updated return statement with other new classNames that make it more consistent and mobile friendly

`search/SearchDetail.js`:
- Removed unused `useNavigate` hook
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Updated `displayPoster` function to change the size of the poster based on whether `isMobile` is true or not
- Added a link that takes the user back to the `SearchBar` component
- Updated return statement with other new classNames that make it more consistent and mobile friendly

`seasons/SeasonDetail.js`:
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Imported new `Episodes` component
- Added logic to update the season poster side based on whether isMobile is true or not
- Defined a new function that displays the season's air date as a more readable string
- Added a link that takes the user back to the `SearchDetail` component they had been viewing
- Updated return statement with other new classNames that make it more consistent and mobile friendly

`seasons/Seasons.js`:
- Imported and defined new `useScreenSize` hook and `isMobile` function
- Removed logic that is now being handled by `useScreenSize` hook
- Defined a function that changes the size of the seasons posters sizes based on whether the user is on mobile or not or if one even exists

### Bug fixes:

None

## Type of change


# How to Test?

Make sure you've followed all installation and setup instructions in the README.

```
git fetch origin nm-styling-cleanup
git checkout nm-styling-cleanup
npm start
```

- [ ] Click `Playlists`
- [ ] Select a playlist
- [ ] Select an episode
- [ ] Click `Back to playlist` link
- [ ] Click `Search Shows`
- [ ] Vaguely search for a show so you get a lot of results
- [ ] Scroll up and down
- [ ] Squish your window so the site goes into mobile view
- [ ] Click on a show
- [ ] Select a season
- [ ] Select an episode
- [ ] Click `Back to season` link
- [ ] Click `Back to show` link
- [ ] Logout if you were logged in
- [ ] Click `Login`
- [ ] Click `Not a user yet?`
- [ ] First name should be auto-focused

# Checklist:
